server/api/utils/pagination.py

```diff
@@ -78,13 +79,16 @@ async def paginate_permission_filtered_request(
                 page_size,
                 **method_kwargs,
             )
+            new_result = await filter_(new_result)
+            result.extend(new_result)
+
             if not pagination_info:
                 # no more results
                 break
+
             last_pagination_info = pagination_info
-            new_result = await filter_(new_result)
-            result.extend(new_result)
-            current_page = pagination_info["page"] + 1
+            current_page = last_pagination_info["page"] + 1
+            page_size = last_pagination_info["page_size"]
```

Comment:
if you don't have a page_size (by human mistake?) you will enter an infinite loop. I understand its unlikely but we should have sanity validation for it